manifests/charts/base/templates/endpoints.yaml

```diff
@@ -3,7 +3,7 @@
 apiVersion: v1
 kind: Endpoints
 metadata:
-  {{- if .Values.pilot.enabled }}
```

Comment:
This doesn't make much sense to me. I would install with `helm install istio/base --set enabled=true`..? What does `enabled` mean -- nothing, without the context of `pilot.`

Reply:
I don't think this really needs to be in `base` anyway, and twiddled with it in https://github.com/istio/istio/pull/53016 - it is not a singleton, as it is revisioned.

Reply:
@fjglira @luksa @bleggett @nshankar13 can you work together to come up with a plan to fix this up (before we ship it in 1.24)?

Reply:
I think the PR @bleggett linked somewhat addresses this, maybe we can continue the discussion there and decide how the values should be organized across the base, istio-discovery, and istiod-remote.

Reply:
https://github.com/istio/istio/pull/53016 should resolve the concerns here.